Nodejs/Product/Nodejs/Project/NodejsFileNode.cs

```diff
@@ -35,14 +35,16 @@ public NodejsFileNode(NodejsProjectNode root, ProjectElement e)
         }
 
         protected override void OnParentSet(HierarchyNode parent) {
-            if (ProjectMgr == null || ProjectMgr.Analyzer == null || !ProjectMgr.ShouldAcquireTypingsAutomatically) {
+            if (ProjectMgr == null || ProjectMgr.Analyzer == null) {
                 return;
             }
 
             if (Url.EndsWith(NodejsConstants.TypeScriptDeclarationExtension, StringComparison.OrdinalIgnoreCase) && Url.IndexOf(@"\typings\", StringComparison.OrdinalIgnoreCase) >= 0) {
-                ProjectMgr.Site.GetUIThread().Invoke(() => {
-                    this.IncludeInProject(true);
-                });
+                if (ProjectMgr.ShouldAcquireTypingsAutomatically) {
```

Comment:
It appears that ShouldAcquireTypingsAutomatically is an unnecessarily expensive operation - instead of performing unnecessary string comparisons, we should just cache the value.